grid/recruitment/list-candidates/profile.supr

```diff
@@ -0,0 +1,209 @@
+        ID of the candidate.
+        """
+        id! string!
+
+        """
+        Job ID
+        ID of a job the candidate is applying for.
+        """
+        jobId! string!
+
+        """
+        Name
+        The name of the candidate.
+        """
+        name! string!
+        
+        """
+        Email
+        Email of the candidate.
+        """
+        email! string!
+
+        """
+        Created At
+        ISO8601 timestamp when was thethe candidate created.
+        """
+        createdAt string
+        
+        """
+        Updated At
+        ISO8601 timestamp when was the candidate lastly updated.
+        """
+        updatedAt string
+      }!
+    ]!
+
+    """
+    Pagination
+    The pagination state.
+    """
+    pagination Pagination
+
+    """
+    Rate limit
+    Rate limit details.
+    """
+    rateLimit RateLimit
+  }
+
+  error Error
+
+  example Success {
+    input {
+      jobId = 'JOB_ID'
+    }
+    result {
+      candidates = [{
+        id = 'CANDIDATE_ID'
+        jobId = 'JOB_ID'
+        name = 'John Doe'
+        email = 'john.doe@acme.com'
+      }]
+    }
+  }
+
+  example Fail {
+    input {
+      jobId = "WRONG_JOB_ID"
+    }
+    error {
+      title = "Not Found"
+      detail = "Job ID \"WRONG_JOB_ID\" do not exist."
+      code = "NotFound"
+    }
+  }
+}
+
+model Pagination {
+  """
+  Previous page
+  Previous page cursor.
+  """
+  previousPage
+
+  """
+  Next page
+  Next page cursor.
+  """
+  nextPage
+}
+
+model Error {
+  """
+  Error Title
+  A short, human-readable summary of the problem type.
+  """
+  title!
+  """
+  Error Detail
+  A human-readable explanation specific to this occurrence of the problem.
+  """
+  detail
+  """
+  Code
+  Error code.
+  """
+  code! ErrorCode!
+  """
+  Rate limit
+  Rate limit details.
+  """
+  rateLimit RateLimit
+}
+
+model RateLimit {
+  """
+  Bucket
+  Different parts of API may have different rate limits.
+  Bucket identifies to which part of API the rate limits apply.
+  """
+  bucket string
+  """
+  Total requests
+  Total requests available in the time window.
+  """
+  totalRequests number
+  """
+  Remaining requests
+  Remaining requests available in the time window.
+  """
+  remainingRequests number
+  """
+  Remaining requests available in percents
+  Remaining requests available in the time window in percents.
+  """
+  remainingRequestsPercentage number
+  """
+  Reset timestamp
+  Timestamp when the rate limits will reset (in  Unix Timestamp format). Available only for providers without rolling rate limit window.
+  """
+  resetTimestamp number
+}
+
+model ErrorCode enum {
+  """
+  Bad request
+  The request was invalid or cannot be otherwise served.
+  """
+  BadRequest
+  """
+  Not found
+  The jobId does not exist.
+  """
+  NotFound
+  """
+  Rate limit reached
+  Returned when a request cannot be served due to the user or app reached max requests quota.
+  """
+  RateLimitReached
+  """
+  Unauthenticated
```

Comment:
IMO We should have use case specific errors in profiles.

All of the recruitment providers requires authentication. Generally authentication error may not be returned depending on the provider.

The client needs to know what kind of errors are returned by the use case. In case of authentication, rate limiting we should provide information what providers are returning it.

This is also related to the topic how to differences in inputs validation. See how I solved it in `recruitment/create-candidate` profile: https://github.com/superfaceai/station/blob/549d7f3224b230f742d1e7b1755491ce6faf74cf/grid/recruitment/create-candidate/profile.supr#L146

Reply:
> Should we have an enum for HTTP errors?

No HTTP errors should not leak from the profiles.